fast-vm

```diff
@@ -162,13 +162,18 @@ handle_multiple_requests () {
 	operation="$1"
 	# VM number is in other position in 'create' operation, change the operation so we don't loose the image_name
 	if [ "$operation" == 'create' ]; then operation="$operation $2"; fi
+	if [ "$operation" == 'scp' ] || [ "$operation" == "keydist" ]; then SERIALIZE=1; fi
 	# get the rest of the command line for calling it later
 	original_arguments=$(echo "$@"|sed -e "s/$operation $vm_number//")
 	pmsg $P_DEBUG "Starting to handle multiple operations\n"
 	all_pids=''
 	for vm in $vm_number; do
 		pmsg $P_DEBUG "Calling operation: ./fast-vm $operation $vm $original_arguments\n"
-		MULTI_OPERATION=1 $fast_vm_bin $operation $vm $original_arguments &
+		if [ -z "$SERIALIZE" ]; then
+			MULTI_OPERATION=1 $fast_vm_bin $operation $vm $original_arguments &
+		else
+			MULTI_OPERATION=1 $fast_vm_bin $operation $vm $original_arguments
+		fi
```

Comment:
Why we serialize multiple operations (specifically the 'scp' and 'keydist')? Is this not working properly in parallel?

Reply:
When a password is required, SERIALIZE allows fast-vm to iterate through the operation one VM at a time.  For keydist, a password will be required; for scp, if there is no ssh key, it would also be required.  Without SERIALIZE, the prompts appear to interfere with each other and the resulting output is confusing.  My thought was this would simplify distributing keys and files to multiple VMs at once.